library/general/src/lib/ui/wizards/layout.rb

```diff
@@ -0,0 +1,286 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+module UI
+  module Wizards
+    # Class to configure a wizard layout
+    #
+    # There are four possible layouts:
+    #
+    # * With a left sidebar which is usually used to place the installation steps.
+    # * With a left tree.
+    # * Without a sidebar/tree and with the title of the dialogs on the left.
+    # * Without a sidebar/tree and with the title of the dialogs on top.
+    #
+    # Moreover, for each layout, a top banner can be added.
+    #
+    # @example
+    #   layout1 = UI::Wizards::Layout.with_steps
+    #   layout2 = UI::Wizards::Layout.with_title_on_left
+    #   layout3 = UI::Wizards::Layout.with_title_on_top
+    #
+    #   layout3.show_banner
+    #
+    #   layout1.open_wizard do
+    #     # ...
+    #   end
+    class Layout
+      Yast.import "Wizard"
+      Yast.import "ProductFeatures"
+
+      # Class to represent a layout mode
+      class Mode
+        STEPS = :steps
+        TREE = :tree
+        TITLE_ON_LEFT = :title_on_left
+        TITLE_ON_TOP = :title_on_top
+
+        private_constant :STEPS, :TREE, :TITLE_ON_LEFT, :TITLE_ON_TOP
+
+        class << self
+          # Mode for a layout with a left sidebar
+          #
+          # @return [Mode]
+          def steps
+            new(STEPS)
```

Comment:
well why methods and always same object. Why it cannot be immutable and something like 
```
STEPS = Mode.new(:steps)
TREE = Mode.new(:tree)
```
etc. ? And if you would like to have methods then `def tree; TREE; done`

Reply:
Finally I have done something similar to your proposal, but saving them to variable instances of the class.